pkg/cmd/generate/cluster_test.go

```diff
@@ -230,6 +231,26 @@ func TestUsers(t *testing.T) {
 			hasNsClusterRole("toolchain-member-operator", "view", configuration.Member.AsSuffix("clusterrole-view-bob-crtadmin")).
 			hasClusterRoleBinding("cluster-monitoring-view", configuration.Member.AsSuffix("clusterrole-cluster-monitoring-view-bob-crtadmin"))
 	})
+
+	t.Run("returns error for invalid username", func(t *testing.T) {
+		for _, username := range invalidUserNames {
+			t.Run("username: "+username, func(t *testing.T) {
+				// given
+				kubeSawAdmins := newKubeSawAdminsWithDefaultClusters(
+					ServiceAccounts(),
+					Users(User("john+user", []string{"12345"}, true, "", permissionsForAllNamespaces...)),
+				)
+				ctx := newAdminManifestsContextWithDefaultFiles(t, kubeSawAdmins)
+				clusterCtx := newFakeClusterContext(ctx, configuration.Host)
+
+				// when
+				err := ensureUsers(clusterCtx, objectsCache{})
+
+				// then
+				require.Error(t, err)
```

Comment:
I usually prefer checking the actual error just to make sure the error message is informative and it's actually the expected error and not something else. Unless it's tricky to hardcode the expected error message in the test.

Reply:
+1  to checking the actual error

Reply:
see https://github.com/kubesaw/ksctl/pull/94#discussion_r1888580456